Cython/Compiler/ExprNodes.py

```diff
@@ -4557,6 +4572,27 @@ def generate_result_code(self, code):
                                   self.index.result()),
                 self.result() if self.type.is_pyobject else None,
                 self.exception_value, self.in_nogil_context)
+        elif self.base.type.is_ctuple:
+            if not self.index.is_literal:
+                self.ctuple_var = code.funcstate.allocate_temp(PyrexTypes.CPtrType(self.type), manage_ref=False)
+    
+                code.putln("switch (%s) {" % index_code)
+    
+                for itr in range(-len(self.base.type.components) if code.globalstate.directives['wraparound'] else 0, len(self.base.type.components)):
+                    code.putln("case %i:" % itr)
+                    code.increase_indent()
+                    code.putln("%s = &%s.f%i;" % (self.ctuple_var, self.base.result(), itr % len(self.base.type.components)))
+                    code.putln("break;")
+                    code.decrease_indent()
```

Comment:
Let's keep the code overhead a little lower here.
```suggestion
                    code.putln(f"case {itr}: {self.ctuple_var} = &{self.base.result()}.f{itr % len(self.base.type.components)}; break")
```
Same below. That also avoids the need to controlling the indentation.

I also suggest using variables for the longer value parts in the f-string above, both to give them names and to make the f-string more readable.

Reply:
You forgot a semicolon.